monitoring/datadog_downtime.py

```diff
@@ -0,0 +1,314 @@
+        argument_spec=dict(
+            api_key=dict(required=True),
+            app_key=dict(required=True),
+            state=dict(required=True, choices=['present', 'updated', 'absent']),
+            scope=dict(required=True, type='list'),
+            start=dict(required=False),
+            end=dict(required=False),
+            message=dict(required=False),
+            active_only=dict(required=False, type='bool'),
+            recurrence=dict(required=False),
+        )
+    )
+
+    # prepare Datadog
+    if not HAS_DATADOG:
+        module.fail_json(msg='datadogpy required for this module')
+
+    options = {
+        'api_key': module.params['api_key'],
+        'app_key': module.params['app_key']
+    }
+
+    initialize(**options)
+    downtimes = find_downtimes_by_scope(module)
+
+    if module.params['state'] == 'present':
+        set_downtime(module, downtimes)
+
+    elif module.params['state'] == 'updated':
+        update_downtimes(module, downtimes)
+
+    elif module.params['state'] == 'absent':
+        cancel_downtimes(module, downtimes)
+
+def set_downtime(module, downtimes):
+    scope = module.params['scope']
+    start = module.params['start']
+    end = module.params['end']
+    message = module.params['message']
+    recurrence = module.params['recurrence']
+    loaded_recurrence = None
+
+    for downtime in downtimes:
+#        print downtime
+        # all scope tags must match - also the number of tags
+        if len(set(scope) - set(downtime['scope'])) == 0:
+            mismatch = False
+
+            # start must be equally defined
+            if start and not downtime['start'] or not start and downtime['start'] :
+                continue
+
+            # message must be equally defined
+            if message and not downtime['message'] or not message and downtime['message'] :
+                continue
+
+            # recurrence must be equally defined
+            if recurrence and not downtime['recurrence'] or not recurrence and downtime['recurrence'] :
+                continue
+
+            # end must be equally defined
+            if end and not downtime['end'] or not end and downtime['end'] :
+                continue
+
+            # start must match
+            if start and int(start) != downtime['start']:
+                continue
+
+            # message must match
+            if message and message and message != downtime['message']:
+                continue
+
+            # recurrence must match
+            if recurrence and recurrence != downtime['recurrence']:
+                continue
+
+            #end must match
+            if end and int(end) != downtime['end']:
+                continue
+
+            #all criteria match so no further action needed
+            module.exit_json(changed=False, msg="Matching downtime already present." % scope)
+
+    # set current time for start if no start given
+    if not start:
+        start = int(time.time())
```

Comment:
the 'time' module is available via the 
``` python
from ansible.module_utils.basic import *
```
but adding a 'import time' is recommended.